### DISCUSSION TITLE
How is the AuthoringMode set

### DISCUSSION THREAD

**Discussion by SeanFarrow** [arcus-azure/arcus.templates#840] [Q&A]
Good afternoon,

I love these templates and was digging in to the implementation from a DevOps/build perspective.

What I cannot find is how the AuthoringMode is ever set to true when building the templates.

I have looked at all the .yml files and have also read the post [here](https://www.codit.eu/blog/arcus-net-project-templates/).

Can someone point me to how/where this is set?

Kind regards
Sean.

**Comment by stijnmoreels** [comment]
Hi @SeanFarrow , great to hear that you love these project templates! 🎉
The `AuthoringMode` you're talking about is something by default present in all .NET templating projects. It allows for certain functionality to be present when building the template, but not building the actual result when the template is run. A good example of this are the project settings files that describe the summary and package information of the template. Those should be available when building the template, but not available when the the template is run to create a new project. Hence, the `AuthoringMode` allows us to do this.
We do not set this flag ourselves, but is available when working with .NET templates.

For more information on .NET templates, I would've indeed recommended the blog post (great that you read that!) and also this .NET templating GitHub repository: https://github.com/dotnet/templating/wiki where there are also a lot of other samples. Microsoft's documentation is rather minimal on this topic, though.

Thanks again for you investment in Arcus! 🎉
Cheers.
